eventstore/views.py

```diff
@@ -20,6 +27,67 @@
     PrebirthRegistrationSerializer,
     PublicRegistrationSerializer,
 )
+from ndoh_hub.utils import TokenAuthQueryString, validate_signature
+
+
+class MessagesViewSet(GenericViewSet):
+    queryset = Message.objects.all()
+    permission_classes = (DjangoModelPermissions,)
+    authentication_classes = (TokenAuthQueryString, TokenAuthentication)
+
+    def create(self, request):
+        validate_signature(request)
+        webhook_type = request.headers.get("X-Turn-Hook-Subscription", None)
+        if webhook_type == "whatsapp":
+            for inbound in request.data.get("messages", []):
+                id = inbound.pop("id")
+                contact_id = inbound.pop("from")
+                type = inbound.pop("type")
+                timestamp = datetime.fromtimestamp(int(inbound.pop("timestamp")))
```

Comment:
If any of these fields are missing, that means that this part of the code will break. We probably want to catch that error and return a 400 status back, letting the client know that they didn't supply one of the required fields.

Reply:
okay will do